BLSQ/iaso
# IA-3629: new endpoint for OrgUnitTypes dropdown

Explain what problem this PR is resolving

Related JIRA tickets : IA-3629

## Self proofreading checklist


## Doc
NA
## Changes

- Add `dropdown` andpoint to `OrgUnitTypeViewSetV2`. The endpoint only returns `id`, `name` and `depth`, the only values used by the front-end when fetching data org unit types dropdowns. This saves a call to `count()` for each org unit type and the computation of a lot of fields
- Update the `useGetIrgUnitTypesDropdown` hook to call the dropdawn endpoint
- Remove and replace duplicates `useOrgUnitTypes` hooks that were used to fetch dropdown data.

## How to test

- All pages/components that display an org unit types dropdown should still have the same behaviour, except that the dropdown should load much faster:
    - FormFormComponent
    - LinksFilters
    - ReviewOrgUnitChangesFilter
    - OrgUnitWriteTypes
    - OrgUnitChangeRequestConfigDialog
    - OrgUnitChangeRequestConfigFilters
    - OrgUnitChildrenFilters
    - OrgUnitTypesDialog
    - InstancesFilterComponent
    - Filter.tsx(mappings)
    - Filters.tsx (Forms)
    - Filters.tsx (Users)
    
- Pay a special attention to:
    - OrgUnitsMap: the type id is used in the `Pane`
    - TableList (org units list view): double check that the list is filtered based on the selected project

## Print screen / video

https://github.com/user-attachments/assets/fdc6b1a7-c091-4aef-a4c3-b36f4f46d2fa




## Notes

This PR ports on main the changes made for IA-3629 on IA-3628_IA-3629_HOTFIX_org_unit_types_and_groups_timeout_bug
